### PR TITLE
Generate C# singletons as non-static classes

### PR DESCRIPTION
**Pull request by raulsntos**
Singleton classes are no longer generated as static classes so they can be inherited.

This fixes mono after #59140 (introduces a class `PhysicsServer3DExtension` that inherits from `PhysicsServer3D` which is a singleton class) and re-enables mono CI.

The disadvantage of this is we now have to access the singleton class using the `Singleton` property everytime, which is cumbersome, but it's consistent with how it works in the engine and godot-cpp (using the `get_singleton` method).

### CODE REVIEW

File: modules/mono/editor/bindings_generator.cpp [hunk 1]
```diff
@@ -1889,7 +1887,7 @@ Error BindingsGenerator::_generate_cs_method(const BindingsGenerator::TypeInterf
 		p_output.append(MEMBER_BEGIN);
 		p_output.append(p_imethod.is_internal ? "internal " : "public ");
 
-		if (p_itype.is_singleton) {
+		if (p_itype.name == "Engine" && p_imethod.name == "get_singleton") {
```
Comment (raulsntos) [inline modules/mono/editor/bindings_generator.cpp:1890]: `Engine.GetSingleton` is still generated as static to avoid an infinite loop.

**Comment by neikeq** [comment]
> The disadvantage of this is we now have to access the singleton class using the `Singleton` property everytime, which is cumbersome, but it's consistent with how it works in the engine and godot-cpp (using the `get_singleton` method).

This is precisely why I made them static classes. It's a pretty big downside. I'll see what can be done about `PhysicsServer3DExtension`.

**Comment by neikeq** [comment]
For now, let's just ignore classes that inherit from singletons.

**Comment by raulsntos** [comment]
Superseeded by #59286

**Pull request closed without merging**